cmake/CeleritasLibrary.cmake

```diff
@@ -86,31 +86,28 @@ define_property(TARGET PROPERTY CELERITAS_CUDA_OBJECT_LIBRARY
 #-----------------------------------------------------------------------------#
 #
 # Internal routine to figure out if a list contains
-# CUDA source code.  Returns TRUE/FALSE in the OUTPUT_VARIABLE
+# CUDA source code.  Returns empty or the list of CUDA files in the var
 #
-function(celeritas_sources_contains_cuda OUTPUT_VARIABLE)
-  set(_contains_cuda FALSE)
+function(celeritas_sources_contains_cuda var)
+  set(_result)
   foreach(_source ${ARGN})
-    get_source_file_property(_iscudafile ${_source} LANGUAGE)
-    if(_iscudafile)
-      if (${_iscudafile} STREQUAL "CUDA")
-        set(_contains_cuda TRUE)
-      endif()
-    else()
+    get_source_file_property(_iscudafile "${_source}" LANGUAGE)
+    if(_iscudafile STREQUAL "CUDA")
+      list(APPEND _result "${_source}")
+    elseif(NOT _iscudafile)
       get_filename_component(_ext "${_source}" LAST_EXT)
       if(_ext STREQUAL ".cu")
-        set(_contains_cuda TRUE)
-        break()
+        list(APPEND _result "${_source}")
       endif()
     endif()
   endforeach()
-  set(${OUTPUT_VARIABLE} ${_contains_cuda} PARENT_SCOPE)
+  set(${var} "${_result}" PARENT_SCOPE)
 endfunction()
 
 #-----------------------------------------------------------------------------#
 #
 # Internal routine to figure out if a target already contains
-# CUDA source code.  Returns TRUE/FALSE in the OUTPUT_VARIABLE
+# CUDA source code.  Returns empty or list of CUDA files in the OUTPUT_VARIABLE
```

Comment:
The current implementation of `celeritas_lib_contains_cuda with a target as argument` does not fullfil the doc in all cases.  If `CELERITAS_CUDA_LIBRARY_TYPE` is already set (likely) it returns `TRUE` (as per the previous intent) rather than the list of files.

Reply:
OK. You can revert/refactor as you see fit. I was just trying to combine the hip/cuda file switching into the same function.